internal/gengapic/gengapic.go

```diff
@@ -95,15 +97,53 @@ func Gen(genReq *plugin.CodeGeneratorRequest) (*plugin.CodeGeneratorResponse, er
 	}
 	g.metadata.LibraryPackage = g.opts.pkgPath
 
+	if !g.opts.omitSnippets {
+		// Initialize the model that will collect snippet metadata.
+		sm, err := snippets.NewMetadata(protoPkg, g.metadata.LibraryPackage, g.opts.pkgName)
+		if err != nil {
+			return &g.resp, err
+		}
+		g.snippetMetadata = sm
+	}
+
 	for _, s := range genServs {
 		// TODO(pongad): gapic-generator does not remove the package name here,
 		// so even though the client for LoggingServiceV2 is just "Client"
 		// the file name is "logging_client.go".
 		// Keep the current behavior for now, but we could revisit this later.
-		outFile := pbinfo.ReduceServName(s.GetName(), "")
-		outFile = camelToSnake(outFile)
+		servName := pbinfo.ReduceServName(s.GetName(), "")
+		outFile := camelToSnake(servName)
 		outFile = filepath.Join(g.opts.outDir, outFile)
 
+		if !g.opts.omitSnippets {
+			defaultHost := proto.GetExtension(s.Options, annotations.E_DefaultHost).(string)
```

Comment:
There are a couple of type casts in this method. Are we posative of thier value or should they be checked to avoid panics?

Reply:
I think the value is quite certain, but good call, will add. Thanks!

Reply:
@noahdietz In addition to this statement, there are 6 existing calls to `proto.GetExtension` that cast to a string without checking `ok`. Are they all good as-is, or do you see any reason to add the checks?

Reply:
The value type of a proto extension will never change (breaking change) and if the annotation is not present, this getter will return the default value (e.g. `""` for default_host). I don't think we need to check the `ok`.